cardano-cli/test/cardano-cli-test/Test/Cli/CheckNodeConfiguration.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Cli.CheckNodeConfiguration where
+
+import           Cardano.Api
+
+import           Control.Monad (forM_)
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Encode.Pretty as Aeson
+import qualified Data.Aeson.Key as Aeson
+import qualified Data.Aeson.KeyMap as Aeson
+import qualified Data.ByteString.Lazy as LBS
+import           Data.List (isInfixOf)
+import qualified Data.Text as Text
+import qualified Data.Yaml as Yaml
+import           GHC.IO.Exception (ExitCode (..))
+import           System.FilePath ((</>))
+
+import           Test.Cardano.CLI.Util (FileSem, bracketSem, execCardanoCLI, execDetailCardanoCLI,
+                   newFileSem)
+
+import           Hedgehog (Property)
+import qualified Hedgehog as H
+import           Hedgehog.Extras (propertyOnce)
+import qualified Hedgehog.Extras as H
+
+-- | Semaphore protecting against locked file error, when running properties concurrently.
+nodeConfigSem :: FileSem
+nodeConfigSem = newFileSem "test/cardano-cli-test/files/input/check-node-configuration/node-config.json"
```

Comment:
This file is not being used anywhere else. We don't need concurrent tests in `cardano-cli` as they run faily quickly so let's avoid the use of a semaphore here.

Reply:
Yes parallelism happens even within a single file. cc @carbolymer